processor/resourcedetectionprocessor/internal/aws/eks/detector.go

```diff
@@ -0,0 +1,210 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package eks
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal"
+)
+
+const (
+	// TypeStr is type of detector.
+	TypeStr = "eks"
+
+	k8sSvcURL = "https://kubernetes.default.svc"
+	/* #nosec */
+	k8sTokenPath      = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+	k8sCertPath       = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+	authConfigmapPath = "/api/v1/namespaces/kube-system/configmaps/aws-auth"
+	cwConfigmapPath   = "/api/v1/namespaces/amazon-cloudwatch/configmaps/cluster-info"
+	timeoutMillis     = 2000
```

Comment:
Nit 1: use time.Duration directly for the const?
```suggestion
	timeout     = 2000 * time.Millisecond
```
It seems like this is just a tiny bit more safe since you won't forget that it is actually milliseconds and not something.

Nit 2: maybe name httpTimeout or httpClientTimeout to better reflect the intent?

Reply:
Definitely agree. Given that the processor handles cancelation/timeouts I'm going to remove the timeout setting from http client completely. (Each detector is given an allotted amount of time to run)